graylog2-web-interface/src/views/logic/SearchResult.js

```diff
@@ -1,17 +1,60 @@
-import _ from 'lodash';
+import { fromJS } from 'immutable';
+import { mapValues, get, compact } from 'lodash';
 import QueryResult from './QueryResult';
-import SearchError from './SearchError';
+import SearchError, { ResultWindowLimitError } from './SearchError';
 
 class SearchResult {
   constructor(result) {
-    this.result = result;
+    this._result = fromJS(result);
 
-    this.results = _.mapValues(result.results, queryResult => new QueryResult(queryResult));
-    this.errors = _.get(result, 'errors', []).map(error => new SearchError(error));
+    this._results = fromJS(mapValues(result.results, queryResult => new QueryResult(queryResult)));
+    this._errors = fromJS(get(result, 'errors', []).map((error) => {
+      if (error.result_window_limit) {
```

Comment:
To distinguish a class on an attribute (other then type) is usually not a good idea.

Is there a reason why `ResultWindowLimitError` is not its own `SearchError` implementation?

But I do not have to strong feelings here.

Reply:
I would also prefer a different approach here. The only idea I have so far is adding a create method for the `SearchError` class to decide which error to create. But I would still just check the attribute inside this method.

Reply:
We've added the `result_window_limit` type for the related error class in the backend, to identify which class to use in the frontend. After talking with @dennisoelkers I've implemented a static create method for the `SearchError` class, to decide which error class we should use.

But I received an error due to the dependency cycle. I can't import the `SearchError` class inside the `ResultWindowLimit` file, while importing the `ResultWIndowLimit` class inside the `SearchError` file. So far I've just used the type attribute to identify the related class inside the above map.